WooCommerce/Classes/ViewRelated/Products/Media/ProductImagesViewController.swift

```diff
@@ -216,15 +216,25 @@ private extension ProductImagesViewController {
             return
         }
         uploadMediaAssetToSiteMediaLibrary(asset: asset)
+        // Commits product image changes automatically.
+        doneButtonTapped()
     }
 }
 
 // MARK: Action handling for device media library picker
 //
 private extension ProductImagesViewController {
     func onDeviceMediaLibraryPickerCompletion(assets: [PHAsset]) {
+        let shouldCommitChanges = assets.isNotEmpty
         defer {
-            dismiss(animated: true, completion: nil)
+            dismiss(animated: !shouldCommitChanges) { [weak self] in
```

Comment:
What do you think about getting rid of the `defer`? 

```swift
        let shouldAnimateMediaLibraryDismissal = assets.isEmpty

        dismiss(animated: shouldAnimateMediaLibraryDismissal) { [weak self] in
            guard let self = self, assets.isEmpty == false else {
                return
            }

            assets.forEach { asset in
                self.uploadMediaAssetToSiteMediaLibrary(asset: asset)
            }

            // Commits product image changes automatically.
            self.doneButtonTapped()
        }
```

This seems more straightforward. And similar to `onWPMediaPickerCompletion`. 🙂

Reply:
yea `defer` is somewhat hard to read. in https://github.com/woocommerce/woocommerce-ios/pull/2235/commits/c1117372c58b521d2b507c737aabb7f5a29be41d, I updated it here and also in WP media picker's callback